File: CMPS12B/prog5/DictionaryTest.c
//-----------------------------------------------------------------------------
// DictionaryTest.c
// Test client for the Dictionary ADT
//-----------------------------------------------------------------------------

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"

#define MAX_LEN 180

int main(int argc, char* argv[]){
   Dictionary A = newDictionary();
   char* k;
   char* v;
   char* word1[] = {"Rachel","Ross","Monica","Chandler","Joey","Phoebe","Janice"};
   char* word2[] = {"Green","Geller","Geller","Bing","Tribbiani","Buffay","Hosenstein"};
   int i;

   for(i=0; i<7; i++){
      insert(A, word1[i], word2[i]);
   }

   printDictionary(stdout, A);

   for(i=0; i<7; i++){
      k = word1[i];
      v = lookup(A, k);
      printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
   }

   // insert(A, "five", "glow"); // error: duplicate keys

   delete(A, "Rachel");
   delete(A, "Monica");
   delete(A, "Janice");

   printDictionary(stdout, A);

   for(i=0; i<7; i++){
      k = word1[i];
      v = lookup(A, k);
      printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
   }

   // delete(A, "one");  // error: key not found

   printf("%s\n", (isEmpty(A)?"true":"false"));
   printf("%d\n", size(A));
   makeEmpty(A);
   printf("%s\n", (isEmpty(A)?"true":"false"));

   freeDictionary(&A);

   return(EXIT_SUCCESS);
}

File: CMPS12B/prog4/DictionaryTest.java
// File name: DictionaryTest.java
// Name: Natalie Petrosian
// Class: 12B
// CruzID: npetrosi
// Description: This tests the methods in Dictionary.java

public class DictionaryTest {
	
	public static void main(String[] args) {
		String v;
	      Dictionary A = new Dictionary();
	      A.insert("1","Rachel");
	      A.insert("2","Ross");
	      A.insert("3","Monica");
	      A.insert("4","Chandler");
	      A.insert("5","Joey");
	      A.insert("6","Phoebe");
	      A.insert("7","Janice");
	      System.out.println(A);

	      v = A.lookup("1");
	      System.out.println("key=1 "+(v==null?"not found":("value="+v)));
	      v = A.lookup("3");
	      System.out.println("key=3 "+(v==null?"not found":("value="+v)));
	      v = A.lookup("7");
	      System.out.println("key=7 "+(v==null?"not found":("value="+v)));
	      v = A.lookup("8");
	      System.out.println("key=8 "+(v==null?"not found":("value="+v)));
	      System.out.println();

	      // A.insert("2","f");  // causes DuplicateKeyException

	      A.delete("1");
	      A.delete("3");
	      A.delete("7");
	      System.out.println(A);

	      // A.delete("8");  // causes KeyNotFoundException

	      System.out.println(A.isEmpty());
	      System.out.println(A.size());
	      A.makeEmpty();
	      System.out.println(A.isEmpty());
	      System.out.println(A);
	}

}
